app/src/main/java/org/mozilla/focus/cfr/CfrMiddleware.kt

```diff
@@ -36,20 +39,30 @@ class CfrMiddleware(private val components: Components) : Middleware<BrowserStat
             }
         }
 
-        if (shouldShowCfrForTrackingProtection(action = action)) {
+        if (shouldShowCfrForTrackingProtection(action = action, browserState = context.state)) {
             components.appStore.dispatch(AppAction.ShowTrackingProtectionCfrChange(true))
         }
     }
 
+    // We shouldn't show CFR for Mozilla pages #6383.
+    private fun isMozillaUrl(browserState: BrowserState) =
+        (
+            browserState.findTabOrCustomTabOrSelectedTab(
+                browserState.selectedTabId
+            )?.content?.url?.toUri()?.hostWithoutCommonPrefixes
+            )?.contains("mozilla")
```

Comment:
Checking for something that **contains** "mozilla" would give false positives.
I think we want to exclude "mozilla" specifically? In which case we should check for equality here?

Reply:
Done